Assets/Scripts/InteractionController.cs

```diff
@@ -0,0 +1,66 @@
+using UnityEngine;
+
+public class InteractionController : MonoBehaviour {
+    [SerializeField]
+    private IInteractable currentTarget;
+
+    public float range = 100;
+    public GameObject FollowTarget;
+
+    private void Update () {
+        RaycastForInteractable ();
+        if (InputManager.Instance.InteractInput && (currentTarget as IInteractable) != null) {
+            (currentTarget as IInteractable).OnInteract ();
+        } else if (InputManager.Instance.CastInput) {
+            if ((currentTarget as ICastable) != null && !MagicManager.Instance.IsSpellActive) {
+                Debug.Log ("Starting Casting");
+                (currentTarget as ICastable).OnInteract ();
```

Comment:
Idk if you know this C# thing but I like it a lot:

//this does a check and a cast, so if it is an iCastable, we can use iCastableInstance
if(curentTarget is ICastable iCastableInstance){
iCastableInstance.OnInteract();
}

Full if statement: 
if(!MagicManager.Instance.IsSpellActive && curentTarget is ICastable iCastableInstance)
{
iCastableInstance.OnInteract();
}

Now that I think about it, the use of as might be more efficient in this case, so I would leave it. but I already typed all this out so I'm just going to comment anyway